frontend: filter comparison by transitions

chaws opened the pull request:
Closes https://github.com/Linaro/squad/issues/566

Add a transitions table that allow users to select which transitions from a comparison to be shown.

- All tests that have exact same results along builds and environments are **not** shown
- By default only *pass-to-fail* and *fail-to-pass* transitions are shown
- If a test has at least one differing result, all its results across builds & envs are displayed
- If the transitions selected didn't pick up any change in a specific environment, that column will **not** be displayed
- If no transitions are selected, the frontend will automatically go back to default (*pass-to-fail* and *fail-to-pass*)
- As boxes are checked/unchecked, the frontend will be updated accordingly
- Check "ignore all transitions" box to go back to regular comparison view

![Screenshot from 2019-08-16 12-59-14](https://user-images.githubusercontent.com/2254825/63184280-601bb280-c02d-11e9-8a8f-79ce1903a214.png)

terceiro commented:
there is a now a merge conflict, please rebase

terceiro commented:
can the transition selection be made more compact?

chaws commented:
@terceiro , rebased and table shrank (see image below). Also I noticed there was a bug with project comparison, which I included a commit in between to fix that. Let me know if you want that fix to be in a separate PR.

![Screenshot from 2019-08-19 20-33-51](https://user-images.githubusercontent.com/2254825/63306539-5c07c300-c2c1-11e9-8268-2e667e403611.png)

terceiro commented:
AIUI "ignore all transitions" should actually say "show all tests".

also I found an issue, see the screenshots below. unless I check "ignore all transitions", no tests are shown.

![1](https://user-images.githubusercontent.com/30027/63381954-17873080-c370-11e9-9650-83ab0b74ff90.png)
![2](https://user-images.githubusercontent.com/30027/63381955-181fc700-c370-11e9-8418-96ac55f46c00.png)

terceiro commented:
FWIW I cannot reproduce this last issue anymore

terceiro commented:
this issue I mentioned last seem to only happen when comparing projects, but not when comparing builds from the same project

chaws commented:
"show all tests" make more sense indeed. I've renamed it and updated the PR

Regarding your issue, it seems that you're not comparing same environments. See

![issue](https://user-images.githubusercontent.com/30027/63381954-17873080-c370-11e9-9650-83ab0b74ff90.png)
